File: spec/python/test_if_values.py
import unittest

from if_values import IfValues

class TestIfValues(unittest.TestCase):
    def test_if_values(self):
        r = IfValues.from_file("src/fixed_struct.bin")

        self.assertEqual(r.codes[0].opcode, 80)
        self.assertEqual(r.codes[0].half_opcode, 40)
        self.assertEqual(r.codes[1].opcode, 65)
        self.assertIsNone(r.codes[1].half_opcode)
        self.assertEqual(r.codes[2].opcode, 67)
        self.assertIsNone(r.codes[2].half_opcode)

File: spec/lua/test_if_values.lua
local luaunit = require("luaunit")

require("if_values")

TestIfValues = {}

function TestIfValues:test_if_values()
    local r = IfValues:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.codes[1].opcode, 80)
    luaunit.assertEquals(r.codes[1].half_opcode, 40)
    luaunit.assertEquals(r.codes[2].opcode, 65)
    luaunit.assertNil(r.codes[2].half_opcode)
    luaunit.assertEquals(r.codes[3].opcode, 67)
    luaunit.assertNil(r.codes[3].half_opcode)
end
